sdk/servicebus/service-bus/test/receiveAndDeleteMode.spec.ts

```diff
@@ -349,60 +352,180 @@ describe("receive and delete", () => {
+      await testDeferredMessage(TestClientType.UnpartitionedSubscription);
     });
 
     it(
       withSessionTestClientType + ": No settlement of the message removes message",
       async function(): Promise<void> {
-        await deferAndReceiveMessage(withSessionTestClientType);
+        await testDeferredMessage(withSessionTestClientType);
       }
     );
   });
+
+  describe("Settlement of deferred msg in ReceiveAndDelete mode", () => {
+    afterEach(async () => {
+      await afterEachTest();
+    });
+
+    let entityNames: EntityName;
+
+    async function testDeferredMessage(testClientType: TestClientType): Promise<ReceivedMessage> {
+      entityNames = await beforeEachTest(testClientType, "peekLock");
+
+      // send message
+      const testMessage = entityNames.usesSessions
+        ? TestMessage.getSessionSample()
+        : TestMessage.getSample();
+      await sender.sendMessages(testMessage);
+
+      // receive and defer the message
+      const [msg] = await receiver.receiveMessages(1);
+      await (msg as ReceivedMessageWithLock).defer();
+      const sequenceNumber = msg.sequenceNumber!;
+      await receiver.close();
+
+      // Receive the deferred message in ReceiveAndDelete mode
+      receiver = await serviceBusClient.test.getReceiveAndDeleteReceiver(entityNames);
+      const [deferredMsg] = await receiver.receiveDeferredMessages(sequenceNumber);
+      if (!deferredMsg) {
+        throw `No message received for sequence number ${sequenceNumber}`;
+      }
+
+      return deferredMsg;
+    }
+
+    const testError = (err: Error, operation: DispositionType): void => {
+      expect(err.message.toLowerCase(), "ErrorMessage is different than expected").includes(
+        `failed to ${operation} the message as the operation is only supported in \'peeklock\' receive mode.`
+      );
+    };
+
+    async function testSettlement(
+      testClienttype: TestClientType,
+      operation: DispositionType
+    ): Promise<void> {
+      const deferredMsg = await testDeferredMessage(testClienttype);
+      // we have to force this cast - the type system doesn't allow this if you've chosen receiveAndDelete
+      // as your lock mode.
+      const msg = deferredMsg as ReceivedMessageWithLock;
+
+      try {
+        if (operation === DispositionType.complete) {
+          await msg.complete();
+        } else if (operation === DispositionType.abandon) {
+          await msg.abandon();
+        } else if (operation === DispositionType.deadletter) {
+          await msg.deadLetter();
+        } else if (operation === DispositionType.defer) {
+          await msg.defer();
+        }
+      } catch (err) {
+        errorWasThrown = true;
+        testError(err, operation);
+      }
+
+      should.equal(errorWasThrown, true, "Error thrown flag must be true");
+    }
+
+    it(noSessionTestClientType + ": complete() throws error", async function(): Promise<void> {
```

Comment:
would it be simpler to just do the [].forEach(() => {}) trick instead?

Like [this](https://github.com/Azure/azure-sdk-for-js/blob/fe6f7d7389db4175cd0184de13cb9207e573ab8c/sdk/servicebus/service-bus/test/internal/sender.spec.ts#L23)

Reply:
See https://github.com/Azure/azure-sdk-for-js/pull/9950#discussion_r454600840